l1-contracts/contracts/bridgehub/Bridgehub.sol

```diff
@@ -0,0 +1,313 @@
+                IL1Bridge(baseTokenBridge[_request.chainId]).bridgehubDepositBaseToken(
+                    _request.chainId,
+                    msg.sender,
+                    token,
+                    _request.mintValue
+                );
+            }
+        }
+
+        address stateTransition = getZkSyncStateTransition(_request.chainId);
+        canonicalTxHash = IZkSyncStateTransition(stateTransition).bridgehubRequestL2Transaction(
+            msg.sender,
+            _request.l2Contract,
+            _request.mintValue,
+            _request.l2Value,
+            _request.l2Calldata,
+            _request.l2GasLimit,
+            _request.l2GasPerPubdataByteLimit,
+            _request.factoryDeps,
+            _request.refundRecipient
+        );
+    }
+
+    /// @notice After depositing funds to the baseTokenBridge, the secondBridge is called
+    /// @notice to return the actual L2 message which is sent to the Mailbox.
+    /// @notice this assumes that either ether is the base token or
+    /// @notice the msg.sender has approved the baseTokenBridge with the mintValue,
+    /// @notice and also the necessary approvals are given for the second bridge.
+    /// @notice This function is great for contract calls to L2, the secondBridge can be any contract.
+    function requestL2TransactionTwoBridges(
+        L2TransactionRequestTwoBridgesOuter calldata _request
+    ) public payable override returns (bytes32 canonicalTxHash) {
+        {
+            address token = baseToken[_request.chainId];
+            // address tokenBridge = baseTokenBridge[_request.chainId];
+
+            if (token == ETH_TOKEN_ADDRESS) {
+                require(msg.value == _request.mintValue + _request.secondBridgeValue, "Bridgehub: msg.value mismatch");
+                // kl todo it would be nice here to be able to deposit weth instead of eth
+                IL1Bridge(baseTokenBridge[_request.chainId]).bridgehubDepositBaseToken{value: _request.mintValue}(
+                    _request.chainId,
+                    msg.sender,
+                    token,
+                    _request.mintValue
+                );
+            } else {
+                require(msg.value == _request.secondBridgeValue, "Bridgehub: msg.value mismatch 2");
+                // note we have to pass token, as a bridge might have multiple tokens.
+                IL1Bridge(baseTokenBridge[_request.chainId]).bridgehubDepositBaseToken(
+                    _request.chainId,
+                    msg.sender,
+                    token,
+                    _request.mintValue
+                );
+            }
+        }
+
+        address stateTransition = getZkSyncStateTransition(_request.chainId);
+        (bool success, bytes memory data) = _request.secondBridgeAddress.call{value: _request.secondBridgeValue}(
+            abi.encodeWithSelector(_request.secondBridgeSelector, _request.chainId, msg.sender, _request.secondBridgeCalldata)
```

Comment:
I would want to:
1) Add restrictions on `_request.secondBridgeSelector` to be `bridgehubDeposit.selector` for now. Even though we can extend it later for others selectors it is the only one should be used now.
2) Add check that `_request.secondBridgeAddress` is a registered bridge

Reply:
I wouldn't do either of this. Maybe we can do 1, (but I would prefer not to). If we do 2.  then people will not be able to write bridges for chains with custom baseTokens. 

Imagine you're doing and NFT or a custom ERC20 bridge. You either have to pass the ownership of the BaseToken to the CustomBridge so that it can handle the deposit via requestL2Transaction, or you can call requestL2TransactionTwoBridges, in which case the CustomBridge only needs to handle its own assets.

Reply:
Ok, I can agree with 2, but want to have 1 implemented. Basically, you make the interface stable and only allow to change selector. Having different selectors gives no principal advantage since custom bridges can implement one function `bridgehubDeposit`, but have branching depending on `_request.secondBridgeCalldata`

Reply:
Ok, 1 done.